### ISSUE TITLE
Services should likely support providing a local coroutine context for invoking their functions

### ISSUE THREAD

**Issue by JakeWharton** [cashapp/zipline#986]
This is mostly to avoid roundtripping through the main thread when you always want to do work on an IO thread, for example.

**Comment by swankjesse** [comment]
For standalone Zipline, there’s not really any bounce on the main thread. Absolutely everything happens on the Zipline dispatcher, and it’s the service implementation’s job to move calls to another dispatcher as desired.

But we could do that move automatically.

**Comment by JakeWharton** [comment]
Yeah I'm not sure I meant to write main thread there. That's what I get for filing a bug in a meeting.

I think what I was trying to convey was the desire to sometimes target the main thread, or perhaps a different dispatcher like IO. And so it's a question of whether we can or should launch with a user-supplied context. And whether that only applies to suspend functions or all functions or what.
